### PR TITLE
Do not load tax information if the order is not editable

### PR DESCRIPTION
**Pull request by toupper**
## Description

This PR addresses this https://github.com/woocommerce/woocommerce-ios/pull/10703#issuecomment-1719859807 from @joshheald, where it's mentioned that we can change the tax rate even if the order is not editable, causing a different total.

It's worth mentioning that that was possible even before, by changing the customer address (this is what we do when changing the tax rate). Even with that, we don't want to make it so prominent by showing a row to edit tax rates.

## Testing instructions

1. Go to Orders
2. Tap on a completed order
3. Make sure that, unlike before, you cannot set a new tax rate to the order.

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/1864060/0b7cdfaf-b1f9-4351-9387-64b5308e6d51

---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by spencertransier** [comment]
@toupper FYI I changed the base branch of this PR to `trunk` instead of `release/15.3` because I deleted that release branch.

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift [hunk 1]
```diff
@@ -1277,6 +1277,13 @@ private extension EditableOrderViewModel {
     }
 
     func configureTaxRates() {
+        // Tax rates are not configurable if the order is not editable.
+        // Even in creation flow we get right at the beginning that the order is not editable, that's why we have to check the flow here as well
```
Comment (joshheald) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift]:
```suggestion
        // Tax rates are not configurable if the order is not editable.
        // In the creation flow orders are initially (incorrectly) reported as not editable, so we have to check the flow here as well
```

**Review by joshheald — APPROVED** [review]
Works as described, thanks.

One suggestion for wording of the comment... it took me a little while to understand it, so I think this is clearer, but if I've misunderstood feel free to change/ignore.

**Comment by toupper** [comment]
Thanks for the suggestion @joshheald ! It's clearer indeed. Committed.

**Pull request merged**